services/paymaster/src/api.rs

```diff
@@ -460,20 +474,42 @@ pub fn get_domain_state_map(domains: Vec<Domain>, mnemonic: &str) -> HashMap<Str
 pub async fn run_server(
     rpc_url_http: String,
     rpc_url_ws: String,
+    ftl_url: Option<String>,
     listen_address: String,
     domains_states: Arc<ArcSwap<HashMap<String, DomainState>>>,
 ) {
-    let rpc = RpcClient::new_with_commitment(
-        rpc_url_http,
-        CommitmentConfig {
-            commitment: CommitmentLevel::Processed,
+    let rpc_http_sender = PooledHttpSender::new(rpc_url_http, RPC_POOL_SIZE);
+
+    let rpc = RpcClient::new_sender(
+        rpc_http_sender,
+        RpcClientConfig {
+            commitment_config: CommitmentConfig {
+                commitment: CommitmentLevel::Processed,
+            },
```

Comment:
nit: I think this reads better
```suggestion
        RpcClientConfig::with_commitment(CommitmentLevel::Processed)
```